src/services/http-client.service.ts

```diff
@@ -94,7 +94,7 @@ export class HttpClientService {
     // tslint:disable-next-line:typedef
     private handleResponse(res: Response, resolve, reject): void {
         if (this.checkBadRequest(res.statusCode)) {
-            this.handleBadRequest(res, res.body, reject);
+            this.handleBadRequest(res.statusCode, res.body, reject);
```

Comment:
It seems that the same thing happens in line 35, could you also handle that case?